onnxruntime/contrib_ops/cpu/quantization/dynamic_quantize_matmul.cc

```diff
@@ -222,9 +222,13 @@ class DynamicQuantizeMatMul final : public MatMulIntegerToFloatBase {
       // Only handle the common case of a 2D weight matrix. Additional matrices
       // could be handled by stacking the packed buffers.
       b_shape_ = tensor.Shape();
-      // TO DO: handle b_shape_.NumDimensions() > 2 and all dimension values but the last two being 1.
-      if (!(b_shape_.NumDimensions() == 2 || (b_shape_.NumDimensions() == 3 && b_shape_[0] == 1))) {
-        can_use_dynamic_quant_mlas_ = false;
+      if (b_shape_.NumDimensions() > 2) {
```

Comment:
thanks for making the dimension check more generic.

the old version set `can_use_dynamic_quant_mlas_` to false if `b_shape_.NumDimensions() < 2` but the new version does not. do we want to allow 0 or 1 dimensions?